eng/common/scripts/Invoke-GitHubAPI.ps1

```diff
@@ -392,28 +387,40 @@ function Update-GitHubIssue {
 
 function Remove-GitHubSourceReferences  {
   param (
-    [Parameter(Mandatory = $true)]
     $RepoOwner,
-    [Parameter(Mandatory = $true)]
     $RepoName,
+    $RepoId = "$RepoOwner/$RepoName",
     [ValidateNotNullOrEmpty()]
     [Parameter(Mandatory = $true)]
     $Ref,
     [ValidateNotNullOrEmpty()]
     [Parameter(Mandatory = $true)]
     $AuthToken
   )
-
   if ($Ref.Trim().Length -eq 0)
   {
     throw "You must supply a valid 'Ref' Parameter to 'Delete-GithubSourceReferences'."
   }
-
-  $uri = "$GithubAPIBaseURI/$RepoOwner/$RepoName/git/refs/$Ref"
+  $uri = "$GithubAPIBaseURI/$RepoId/git/refs/$Ref"
 
   return Invoke-RestMethod `
           -Method DELETE `
           -Uri $uri `
           -Headers (Get-GitHubApiHeaders -token $AuthToken) `
           -MaximumRetryCount 3
-}
\ No newline at end of file
+}
+
+
+function Get-GithubReferenceCommitDate($commitUrl) {
+  try {
+    $commitResponse = Invoke-RestMethod $commitUrl
+    if (!$commitResponse.committer -or !$commitResponse.committer.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    return $commitData.committer.date
+  }
+  catch {
+    LogError "Failed to invoke $commitUrl with exception:`n$_"
+    exit 1
```

Comment:
I don't think we want to exit 1 here we should just return null and let the caller decide what they want to do.